swag/blockchain/synced_blockchain.py

```diff
@@ -44,11 +42,11 @@ async def append(self, block):
         if isinstance(block, YfuGenerationBlock):
```

Comment:
I strongly dislike the fact we hardcode a special handling of `YfuGeneration`
But of course, we can't base64-encode the image in the json message since it's to big, and it would be stupid anyway since we mustn't upload the image to discord at every bot startup.

Thinking hard about it, I believe a better way would be either a generic `Block` method to handle Files (meh), or a special `AssetUpload` block (handled the same way as it is now; buuut 1. it's more generic 2. if we dump the blockchain in CBOR and need to recover from that, I believe it's easier to reupload assets and fix urls)

What do you think?

Reply:
I share your dissaprovement of the special handleling of `YfuGenerationBlock`. I just did that to get something runnable quickly.

I like the idea of a `AssetUpload` Block. It can be usefull for the future.

We're gonna have some work to well connect one block with an AssetUpload block.

Futhermore, I like the idea to split the image upload to the `YfuGenerationBlock` in order to be able to generate Yfu even if the computer which is running the bot don't downloaded the avatar pictures

Reply:
Leave it to me, I'll do it
Pretty good job anyway!